sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store.asciidoc

```diff
@@ -0,0 +1,588 @@
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `ItemsPerWorkItem`.
+Properties may also provide a hint for optimization purposes
+xref:optimization_hints[optimization_hints].
+
+=== Data Placement
+
+anchor:data_placement[]
+
+To specify a correct data placement for placing of resulting data
+there is a enum
+(proposed also in link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]):
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Properties:
+enum class data_placement_enum {
+  blocked,
+  striped
+};
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+Data placement layout example on group_load:
+
+* ElementsPerWorkItem = 4
+* 3 work-items in the group
+* input is: in_iter[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::data_placement_enum::blocked`.
+
+|===
+|Work-item id|Output stored in a fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::data_placement_enum::striped`.
+
+|===
+|Work-item id|Output stored in a fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+This extention adds a property that satisfies
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[SYCL Properties Extension]
+requirements to identify data_placement similar to
+link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]
+`input_data_placement` and `output_data_placement`:
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct data_placement_key {
+  template <data_placement_enum Type>
+  using value_t =
+      property_value<data_placement_key,
+                     std::integral_constant<data_placement_enum, Type>>;
+};
+
+template <data_placement_enum Type>
+inline constexpr data_placement_key::value_t<Type> data_placement;
+
+inline constexpr data_placement_key::value_t<data_placement_enum::blocked> data_placement_blocked;
+inline constexpr data_placement_key::value_t<data_placement_enum::striped> data_placement_striped;
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+Specifies data layout used in group_load/store for `sycl::vec` or fixed-size
+arrays functions.
+
+Example:
+`group_load(g, input, output_span, data_placement_blocked);`
+
+=== Optimization Hints
+
+anchor:optimization_hints[]
+
+==== Contiguous memory
+
+As `InputIteratorT` and `OutputIteratorT` are permitted to be random access
+iterators they are not guaranteed to be contiguous.
+The following property is introduced to be used
+as a hint to the implementation for optimization purposes:
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+struct contiguous_memory_hint {
```

Comment:
```suggestion
struct contiguous_memory_key {
```

The naming convention for properties is that the struct has the suffix "_key" and the name of the property is the same but without the suffix.  Therefore, either name this `contiguous_memory_hint_key` / `contiguous_memory_hint` or  `contiguous_memory_key` / `contiguous_memory`,

As @Pennycook mentioned elsewhere, this is an assertion, not a hint, so I think we should not use "hint" in the name.  Therefore, if you prefer the longer name, it should be like `contiguous_memory_assertion_key`.